firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/I2C/LTC4015/Sensor.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ *      Author: Rohan Purohit
+ */
+
+#pragma once
+
+#include "Device.h"
+#include "Pufferfish/Application/Alarms.h"
+#include "Pufferfish/Driver/Initializable.h"
+#include "Pufferfish/HAL/Interfaces/Time.h"
+
+namespace Pufferfish::Driver::I2C::LTC4015 {
+
+/**
+ * State management for LTC4015, without I/O.
+ * This is basically a Moore machine, so the state consists of the
+ * next action to take, along with the current time.
+ */
+class StateMachine {
+ public:
+  // More actions will be added later
+  enum class Action { initialize, measure };
+
+  [[nodiscard]] Action update(uint32_t current_time_us);
+
+ private:
+  Action next_action_ = Action::initialize;
+  uint32_t current_time_us_ = 0;
```

Comment:
Do you still need `current_time_us_` in `StateMachine`? It doesn't look like it's being used anywhere.

Reply:
Yep not using it, cleaned up the driver